medusa/classes.py

```diff
@@ -336,13 +336,13 @@ def select_series(self, all_series):
                 search_term = self.config['searchterm']
                 # try to pick a show that's in my show list
                 for cur_show in all_series:
-                    if cur_show in search_results:
+                    if [result for result in search_results if str(cur_show['id']) == str(result['id'])]:
```

Comment:
Can we really compare the ids? What if the you have the show with the same id from another indexer?

Reply:
I tested this. And as you can see in the screenshot above, it will give back both shows.

Reply:
This is run in the context of one indexer search